components/log4j/org.wso2.carbon.analytics.shared.data.agents.log4j/src/main/java/org/wso2/carbon/analytics/shared/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -197,47 +207,55 @@ public void close() {
-        Logger logger = Logger.getLogger(event.getLoggerName());
-        TenantAwareLoggingEvent tenantEvent;
-        if (event.getThrowableInformation() != null) {
-            tenantEvent = new TenantAwareLoggingEvent(event.fqnOfCategoryClass, logger, event.timeStamp,
-                    event.getLevel(), event.getMessage(), event.getThrowableInformation().getThrowable());
-        } else {
-            tenantEvent = new TenantAwareLoggingEvent(event.fqnOfCategoryClass, logger, event.timeStamp,
-                    event.getLevel(), event.getMessage(), null);
-        }
-        tenantId = AccessController.doPrivileged(new PrivilegedAction<Integer>() {
-            public Integer run() {
-                return CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        // Checking the configuration context service for wait DAS appender until proper server initialization.
+        if (LogAppenderServiceValueHolder.getConfigurationContextService() != null) {
+            if (dataPublisher == null) {
+                publisherInitializer();
             }
-        });
-        if (tenantId == MultitenantConstants.INVALID_TENANT_ID) {
-            String tenantDomain = TenantDomainSetter.getTenantDomain();
-            if (tenantDomain != null && !tenantDomain.equals("")) {
-                try {
-                    tenantId = getTenantIdForDomain(tenantDomain);
-                } catch (UserStoreException e) {
-                    System.err.println("Cannot find tenant id for the given tenant domain.");
-                    e.printStackTrace();
+
+            Logger logger = Logger.getLogger(event.getLoggerName());
+            TenantAwareLoggingEvent tenantEvent;
+            if (event.getThrowableInformation() != null) {
+                tenantEvent = new TenantAwareLoggingEvent(event.fqnOfCategoryClass, logger, event.timeStamp,
+                        event.getLevel(), event.getMessage(), event.getThrowableInformation().getThrowable());
+            } else {
+                tenantEvent = new TenantAwareLoggingEvent(event.fqnOfCategoryClass, logger, event.timeStamp,
+                        event.getLevel(), event.getMessage(), null);
+            }
+            tenantId = AccessController.doPrivileged(new PrivilegedAction<Integer>() {
+                public Integer run() {
+                    return CarbonContext.getThreadLocalCarbonContext().getTenantId();
+                }
+            });
+            if (tenantId == MultitenantConstants.INVALID_TENANT_ID) {
+                String tenantDomain = TenantDomainSetter.getTenantDomain();
+                if (tenantDomain != null && !tenantDomain.equals("")) {
+                    try {
+                        tenantId = getTenantIdForDomain(tenantDomain);
+                    } catch (UserStoreException e) {
+                        System.err.println("Cannot find tenant id for the given tenant domain.");
+                        e.printStackTrace();
```

Comment:
Hi Manorama,

Reason for  use  "e.printStackTrace"  instead of "Log" , since this log appender has cyclic dependency with log4j when we log the error it case to literately trigger its own logs.

Thanks
SajithD